File: SECRET_PROJECT_WEB/src/types/ChannelMessage/ChannelMessage.ts
import type {
  Message,
  MessageReaction,
  RepliedMessage,
} from "../Message/Message";

export const isChannelMessage = (
  message: Message | ChannelMessage
): message is ChannelMessage => {
  return typeof message === "object" && "file" in message;
};

export type ChannelMessage = {
  id: string;
  content?: string;
  sentAt: string;
  senderId: string;
  channelId: string;
  file?: File;
  repliedMessage?: RepliedMessage;
  reactions?: MessageReaction[];
};

type File = {
  fileUrl?: string;
  fileType?: string;
  fileName?: string;
};

File: SECRET_PROJECT_SERVER/Secret_Project_Backend/Secret_Project_Backend/Models/ChannelMessage.cs
namespace Secret_Project_Backend.Models
{
    public class ChannelMessage
    {
        public Guid Id { get; set; }
        public string? Content { get; set; }
        public Guid ChannelId { get; set; }
        public virtual Channel Channel { get; set; }
        public string SenderId { get; set; }
        public virtual ApplicationUser Sender { get; set; }
        public DateTime SentAt { get; set; }
        public Guid ChannelFileId { get; set; }
        public virtual ChannelFile? ChannelFile { get; set; }
        public Guid? RepliedId { get; set; }
        public ChannelMessage? RepliedChannelMessage { get; set; }
        public virtual ICollection<Reaction>? Reactions { get; set; }
    }
}
